README.md

```diff
@@ -10,43 +10,220 @@
-    Try it Now
-    <img alt="Binder" valign="bottom" height="25px"
-    src="https://mybinder.org/badge_logo.svg"
-    />
-  </a>
-</h3>
+**You can try IDOM now in a Jupyter Notebook:**
+<a
+  target="_blank"
+  href="https://mybinder.org/v2/gh/idom-team/idom-jupyter/main?filepath=notebooks%2Fintroduction.ipynb">
+  <img
+    alt="Binder"
+    valign="bottom"
+    height="21px"
+    src="https://mybinder.org/badge_logo.svg"/>
+</a>
 
-Click the badge above to get started! It will take you to a [Jupyter Notebooks](https://jupyter.org/)
-hosted by [Binder](https://mybinder.org/) with some great examples.
 
-### Or Install it Now
+# Install Django IDOM
 
 ```bash
 pip install django-idom
 ```
 
 # Django Integration
 
-This version of IDOM can be directly integrated into Django. For example
+To integrate IDOM into your application you'll need to modify or add the following files to `your_app`:
+
+```
+your_app/
+├── asgi.py
+├── components.py
+├── idom.py
+├── settings.py
+├── templates/
+│   ├── your-template.html
+└── urls.py
+```
+
+## `asgi.py`
+
+To start, we'll need to use [`channels`](https://channels.readthedocs.io/en/stable/) to
+create a `ProtocolTypeRouter` that will become the top of our ASGI application stack.
+Under the `"websocket"` protocol, we'll then add a path for IDOM's websocket consumer
+using `django_idom_websocket_consumer_path`. If you wish to change the route where this
+websocket is served from see the [settings](#configuration-options).
+
+```python
+
+import os
+
+from django.core.asgi import get_asgi_application
+
+from django_idom import django_idom_websocket_consumer_path
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "test_app.settings")
+
+# Fetch ASGI application before importing dependencies that require ORM models.
+http_asgi_app = get_asgi_application()
+
+from channels.routing import ProtocolTypeRouter, URLRouter
+
+application = ProtocolTypeRouter(
+    {
+        "http": http_asgi_app,
+        "websocket": URLRouter(
+          # add a path for IDOM's websocket
+          [django_idom_websocket_consumer_path()]
+        ),
+    }
+)
+```
+
+## `components.py`
+
+This is where, by a convention similar to that of
+[`views.py`](https://docs.djangoproject.com/en/3.2/topics/http/views/), you'll define
+your [IDOM](https://github.com/idom-team/idom) components. Ultimately though, you should
+feel free to organize your component modules you wish.
+
+```python
+import idom
+
+@idom.component
+def Hello(name):  # component names are camelcase by convention
+    return idom.html.h1(f"Hello {name}!")
+```
+
+## `idom.py`
+
+This file is automatically discovered by `django-idom` when scanning the list of
+[`INSTALLED_APPS`](https://docs.djangoproject.com/en/3.2/ref/settings/#std:setting-INSTALLED_APPS).
+All apps that export components will contain this module.
+
+Inside this module must be a `components` list that is imported from
+[`components.py`](#components.py):
+
+```python
+from .components import Hello
```

Comment:
This convention actually comes from React, and is a style that I'm going to try and encourage since it will allow this [Flake8 Plugin](https://github.com/idom-team/flake8-idom) to enforce the "rules of hooks".

Will add a second component though to better demo the list.